application-services/custom/camera-management/appcamera/evam.go

```diff
@@ -218,6 +223,99 @@ func (app *CameraManagementApp) getPipelineStatus(deviceName string) (interface{
+		return false, fmt.Errorf("type received %T is not a SystemEvent", data)
+	}
+
+	app.lc.Debugf("Received system event: %v", systemEvent)
+
+	if systemEvent.Type != common.DeviceSystemEventType {
+		app.lc.Error("System event type is not " + common.DeviceSystemEventType)
+		return false, fmt.Errorf("system event type is not " + common.DeviceSystemEventType)
+	}
+
+	device := dtos.Device{}
+	err := systemEvent.DecodeDetails(&device)
+	if err != nil {
+		app.lc.Errorf("failed to decode device details: %v", err)
+		return false, fmt.Errorf("failed to decode device details: %v", err)
+	}
+
+	switch systemEvent.Action {
+	case common.DeviceSystemEventActionAdd:
+		if err = app.startDefaultPipeline(device); err != nil {
+			return false, err
+		}
+	case common.DeviceSystemEventActionDelete:
+		// stop any running pipelines for the deleted device
+		if info, found := app.getPipelineInfo(device.Name); found {
+			if err = app.stopPipeline(device.Name, info.Id); err != nil {
+				return false, fmt.Errorf("error stopping pipleline for device %s, %v", device.Name, err)
+			}
+		}
+	default:
+		app.lc.Debugf("System event action %s is not handled", systemEvent.Action)
+	}
+
+	return false, nil
+}
+
+func (app *CameraManagementApp) startDefaultPipeline(device dtos.Device) error {
+	pipelineRunning := app.isPipelineRunning(device.Name)
+
+	if pipelineRunning {
+		app.lc.Debugf("pipeline is already running for device %s", device.Name)
+		return nil
+	} else {
+		app.lc.Debugf("pipeline is not running for device %s", device.Name)
+	}
+
+	if app.config.AppCustom.DefaultPipelineName == "" || app.config.AppCustom.DefaultPipelineVersion == "" {
+		app.lc.Warnf("no default pipeline name/version specified, skip starting pipeline for device %s", device.Name)
+		return nil
+	}
+
+	startPipelineRequest := StartPipelineRequest{
+		PipelineName:    app.config.AppCustom.DefaultPipelineName,
+		PipelineVersion: app.config.AppCustom.DefaultPipelineVersion,
+	}
+
+	protocol, ok := device.Protocols["Onvif"]
+	if ok {
+		app.lc.Debugf("Onvif protocol information found for device: %s message: %v", device.Name, protocol)
+		profileResponse, err := app.getProfiles(device.Name)
+		if err != nil {
+			app.lc.Errorf("failed to get profiles for device %s, message: %v", device.Name, err)
+			return err
+		}
+
+		app.lc.Debugf("Onvif profile information found for device: %s message: %v", device.Name, profileResponse)
+		startPipelineRequest.Onvif = &OnvifPipelineConfig{
+			ProfileToken: string(profileResponse.Profiles[0].Token),
+		}
+	} else if _, ok := device.Protocols["USB"]; ok {
+		app.lc.Debugf("Usb protocol found for device: %s", device.Name)
+		startPipelineRequest.USB = &USBStartStreamingRequest{}
+	}
+
+	app.lc.Debugf("Starting default pipeline for device %s", device.Name)
+	if err := app.startPipeline(device.Name, startPipelineRequest); err != nil {
+		app.lc.Errorf("Pipeline failed to start for device %s, message: %v", device.Name, err)
+		return err
```

Comment:
```suggestion
		return fmt.Errorf("Pipeline failed to start for device %s, message: %v", device.Name, err)
```

Reply:
I changed the 'pipeline' to lowercase due to a static analysis comment.. let me know if you want it uppercase.

Reply:
@brianointel , yes that also is needed.